simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsDirBlockStreamManager.java

```diff
@@ -106,32 +106,33 @@ public Block getNextBlock() {
     }
 
     private void loadBlocks() throws IOException, ParseException {
-        Path rootPath = Path.of(rootFolder);
+        final Path rootPath = Path.of(rootFolder);
 
-        try (Stream<Path> blockDirs = Files.list(rootPath).filter(Files::isDirectory)) {
-            List<Path> sortedBlockDirs =
+        try (final Stream<Path> blockDirs = Files.list(rootPath).filter(Files::isDirectory)) {
+            final List<Path> sortedBlockDirs =
                     blockDirs.sorted(Comparator.comparing(Path::getFileName)).toList();
 
-            for (Path blockDirPath : sortedBlockDirs) {
-                List<BlockItem> parsedBlockItems = new ArrayList<>();
+            for (final Path blockDirPath : sortedBlockDirs) {
+                final List<BlockItem> parsedBlockItems = new ArrayList<>();
 
-                try (Stream<Path> blockItems =
+                try (final Stream<Path> blockItems =
                         Files.list(blockDirPath).filter(Files::isRegularFile)) {
-                    List<Path> sortedBlockItems =
+                    final List<Path> sortedBlockItems =
                             blockItems
                                     .sorted(
                                             Comparator.comparing(
                                                     BlockAsDirBlockStreamManager
                                                             ::extractNumberFromPath))
                                     .toList();
```

Comment:
Ugh, this is just super ugly with the excessive wrapping.
If we can add a temporary variable or two to fix that, it would be good.

Reply:
✅ I improved this, extracted the comparator to be more readable.